### PR TITLE
docs: fix dead links and update Windows build instructions

### PR DESCRIPTION
**Pull request by MartinMa**
##### Description of Change
Fix dead links (to macOS build instructions) and improve instructions for building on Windows 10.

/cc @nornagon @codebytere @jkleinsc

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes

Notes: no-notes

### CODE REVIEW

File: docs/development/build-instructions-windows.md [hunk 1]
```diff
@@ -8,11 +8,22 @@ Follow the guidelines below for building Electron on Windows.
 * Visual Studio 2017 15.7.2 or higher - [download VS 2017 Community Edition for
   free](https://www.visualstudio.com/vs/)
 * [Python 2.7](http://www.python.org/download/releases/2.7/)
+  * Note, that you will need at least version 2.7.10. Right now `depot_tools`
+  still comes with version 2.7.6, which will cause the `gclient` command to fail
+  (see https://crbug.com/868864). Other than explained in the GN build
```
Comment (MarshallOfSound) [inline docs/development/build-instructions-windows.md]: "Contrary to the `depot_tools` setup instructions linked below..."

File: docs/development/build-instructions-windows.md [hunk 2]
```diff
@@ -8,11 +8,22 @@ Follow the guidelines below for building Electron on Windows.
...
+  instructions linked below, you have to configure your **PATH** so that your
+  locally installed Python comes before the `depot_tools` folder.
 * [Node.js](https://nodejs.org/download/)
 * [Git](http://git-scm.com)
 * [Debugging Tools for Windows](https://msdn.microsoft.com/en-us/library/windows/hardware/ff551063.aspx)
   if you plan on creating a full distribution since `symstore.exe` is used for
   creating a symbol store from `.pdb` files.
+  * If you run into build problems using the latest Windows SDK, try to install
```
Comment (MarshallOfSound) [inline docs/development/build-instructions-windows.md]: There is an exact version of the SDK that Chromium is known to build against and we have on our CI machines, that version should be documented here cc @jkleinsc

Comment (MartinMa) [reply docs/development/build-instructions-windows.md]: On [Checking out and Building Chromium for Windows](https://chromium.googlesource.com/chromium/src/+/master/docs/windows_build_instructions.md#visual-studio) they say that Windows SDK version `10.0.17134` is needed. I am using the older version `10.0.15063.468` which seems to work, too.

Comment (MarshallOfSound) [reply docs/development/build-instructions-windows.md]: @MartinMa That doc is for Chromium master, Electron master is back on Chromium 67 and might have different reqs

Comment (alexeykuzmin) [reply docs/development/build-instructions-windows.md]: > I am using the older version 10.0.15063.468 which seems to work, too.

Docs for Chromium 67 mention exactly `10.0.15063`.

https://chromium.googlesource.com/chromium/src/+/67.0.3396.127/docs/windows_build_instructions.md#visual-studio

**Comment by MartinMa** [comment]
Thanks, I updated `build-instructions-windows.md` accordingly and merged my last three commits into one.

The text now mentions exactly Windows SDK version `10.0.15063.468`. I also added a note about the necessity to install Python for Windows (pywin32) Extensions. Otherwise the ninja build will fail early with a `ImportError: No module named win32file` kind of message. Alternatively it's possible to switch back to the depot_tools supplied Python for building. But explaining to dwitch back and forth between different Python versions (local and depot_tools) seems more difficult.

**Comment by MartinMa** [comment]
@MarshallOfSound Thanks, I didn't see that. I updated and rebased my fork and resolved all conflicts (see [this commit](https://github.com/MartinMa/electron/commit/3b41118be79decd3417a6e45487dee8a23eca6cb)). Weirdly enough, the changes don't seem to be picked up by GitHub in this PR.

Any idea on how to resolve this? Maybe create a new PR?

**Comment by MartinMa** [comment]
@MarshallOfSound Thanks, I didn't see that. I updated and rebased my fork and resolved all conflicts (see [this commit](https://github.com/MartinMa/electron/commit/3b41118be79decd3417a6e45487dee8a23eca6cb)). Weirdly enough, the changes don't seem to be picked up by GitHub in this PR.

Any idea on how to resolve this? Maybe create a new PR?

**Comment by MartinMa** [comment]
@MarshallOfSound Thanks, I didn't see that. I updated and rebased my fork and resolved all conflicts (see [this commit](https://github.com/MartinMa/electron/commit/3b41118be79decd3417a6e45487dee8a23eca6cb)). Weirdly enough, the changes don't seem to be picked up by GitHub in this PR.

Any idea on how to resolve this? Maybe create a new PR?

**Comment by MartinMa** [comment]
@MarshallOfSound Thanks, I didn't see that. I updated and rebased my fork and resolved all conflicts (see [this commit](https://github.com/MartinMa/electron/commit/3b41118be79decd3417a6e45487dee8a23eca6cb)). Weirdly enough, the changes don't seem to be picked up by GitHub in this PR.

Any idea on how to resolve this? Maybe create a new PR?

**Pull request merged**